woocommerce/woocommerce-gateway-stripe
# Fix unable to find xfs user error when running wordpress:cli docker commands

## Changes proposed in this Pull Request:

This PR fixes the following error when running docker commands as a specific user:

`docker: Error response from daemon: unable to find user xfs: no matching entries in passwd file.`

![Screenshot 2024-07-05 at 5 28 55 PM](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/407542/e244036e-8a4e-46ad-ab01-f00ac55a9599)

--

According to the official docs (https://hub.docker.com/_/wordpress) using `--user 33:33` might be needed when running against an existing wordpress/debian image:

![Screenshot 2024-07-05 at 5 15 56 PM](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/407542/e4e98f43-c89e-4857-95ca-fca6e1b4486e)

We use `wordpress:php7.4` which is based on `debian:11-slim`, however just removing `--user xfs` fixes the problem.

## Testing instructions

1. Checkout the `develop` branch and make sure docker is running
3. Run `npm run down` (or manually remove the Stripe dev containers)
4. Run `npm run up` to recreate them
5. Check the repeating `unable to find user` error is present
6. Stop the execution with (multiple) `Ctrl+C`; otherwise, it keeps re-trying indefinitely.
7. Checkout the branch `chore/fix-docker-image-username`
8. Run `npm run down` (or manually remove the Stripe dev containers)
9. Run `npm run up` to recreate them
10. Check that the error is not raised this time
11. Run `npm run wp config get`
12. Check that no error is raised and that the config table is returned.
